services/motion/builtin/builtin.go

```diff
@@ -167,19 +166,58 @@ func (ms *builtIn) Move(
 	return true, nil
 }
 
-// MoveOnMap will move the given component
+// MoveOnMap will move the given component to the given destination on the slam map generated from a slam service specified by slamName.
+// Bases are the only component that supports this.
 func (ms *builtIn) MoveOnMap(
 	ctx context.Context,
 	componentName resource.Name,
 	destination spatialmath.Pose,
 	slamName resource.Name,
 	extra map[string]interface{},
 ) (bool, error) {
+	operation.CancelOtherWithLabel(ctx, "motion-service")
+
+	// get the SLAM Service from the slamName
 	slamService, ok := ms.slamServices[slamName]
 	if !ok {
 		return false, errors.Wrap(resource.NewNotFoundError(slamName), "motion service missing weak dependency")
 	}
-	_ = slamService
+
+	// create a KinematicBase from the componentName
+	b, ok := ms.components[componentName]
+	if !ok {
+		return false, fmt.Errorf("only Base components are supported for MoveOnMap: could not find an Base named %v", componentName)
+	}
+	kw, ok := b.(base.KinematicWrappable)
+	if !ok {
+		return false, fmt.Errorf("cannot move base of type %T because it is not KinematicWrappable", b)
+	}
+	kb, err := kw.WrapWithKinematics(ctx, slamService)
+	if err != nil {
+		return false, err
+	}
+
+	// get current position
+	inputs, err := kb.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+	ms.logger.Infof("base position: %v", inputs)
```

Comment:
[minor] these seem more suited to debug-level logs, less noise.

Reply:
Ok will change